exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -387,3 +409,25 @@ func (r *ClusterResourceSetReconciler) clusterToClusterResourceSet(o handler.Map
 	}
 	return result
 }
+
+// resourceToClusterResourceSet is mapper function that maps resources to ClusterResourceSet
+func (r *ClusterResourceSetReconciler) resourceToClusterResourceSet(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+
+	crsList := &addonsv1.ClusterResourceSetList{}
+	if err := r.Client.List(context.Background(), crsList, client.InNamespace(o.Meta.GetNamespace())); err != nil {
+		return nil
+	}
+
+	for _, crs := range crsList.Items {
+		for _, resource := range crs.Spec.Resources {
+			if resource.Kind == o.Object.GetObjectKind().GroupVersionKind().Kind && crs.Name == o.Meta.GetName() {
+				name := client.ObjectKey{Namespace: o.Meta.GetNamespace(), Name: crs.Name}
+				result = append(result, ctrl.Request{NamespacedName: name})
+				break
+			}
+		}
+	}
+
+	return result
+}
```

Comment:
Can't we use the `o` object's OwnerReferences to enqueue requeuests for CRS resources associated with it?

Reply:
We cannot do that because newly created resources will not have owner ref until CRS is reconciled.